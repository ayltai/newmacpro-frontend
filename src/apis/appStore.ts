import { createApi, fetchBaseQuery, } from '@reduxjs/toolkit/query/react';

import { API_MAX_RETRIES, } from '../constants';
import { AppStoreSearchResponse, Package, } from '../types';

export const appStoreApi = createApi({
    reducerPath : 'appStoreApi',
    baseQuery   : fetchBaseQuery({
        baseUrl : 'https://search-itunes.vercel.app',
    }),
    endpoints   : builder => ({
        search : builder.query<Package[], string>({
            query             : keyword => `/?media=software&entity=macSoftware&limit=3&term=${keyword}`,
            transformResponse : (response : AppStoreSearchResponse) => (response?.results || []).map(result => ({
                id              : String(result.trackId),
                displayName     : String(result.trackName),
                description     : result.description,
                version         : result.version,
                logoUrls        : [
                    result.artworkUrl60,
                    result.artworkUrl100,
                    result.artworkUrl512,
                ].filter(url => url !== undefined) as string[],
                screenshotUrls  : result.screenshotUrls,
                provider        : result.sellerName ?? result.artistName,
                website         : result.trackViewUrl,
                userRating      : result.averageUserRatingForCurrentVersion,
                userRatingCount : result.userRatingCountForCurrentVersion,
                advisoryRating  : result.contentAdvisoryRating,
                price           : result.formattedPrice,
                source          : 'App Store',
            })),
            extraOptions      : {
                maxRetries : API_MAX_RETRIES,
            },
        }),
    }),
});

export const { useSearchQuery, } = appStoreApi;
