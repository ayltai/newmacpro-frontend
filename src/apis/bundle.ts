import { createApi, fetchBaseQuery, } from '@reduxjs/toolkit/query/react';

import { API_MAX_RETRIES, } from '../constants';
import type { Bundle, Package, } from '../types';

export type BundleGetParams = {
    bundleId? : string,
    idToken?  : string,
};

export type BundleCreateParams = {
    bundle  : Bundle,
    idToken : string,
};

export type BundleUpdateParams = {
    bundle  : Bundle,
    idToken : string,
};

export type BundleDeleteParams = {
    bundleId : string,
    idToken  : string,
};

const transformPackages = (packages : any[]) : Package[] => packages.map(pkg => ({
    id               : pkg.id,
    displayName      : pkg.displayName,
    description      : pkg.description,
    version          : pkg.version,
    license          : pkg.license,
    logoUrls         : pkg.logoUrls?.split('|'),
    screenshotUrls   : pkg.screenshotUrls?.split('|'),
    provider         : pkg.provider,
    website          : pkg.website,
    monthlyDownloads : Number(pkg.monthlyDownloads),
    userRating       : Number(pkg.userRating),
    userRatingCount  : Number(pkg.userRatingCount),
    advisoryRating   : pkg.advisoryRating,
    price            : pkg.price,
    source           : pkg.source,
    value            : pkg.value,
}));

export const bundleApi = createApi({
    reducerPath : 'bundleApi',
    baseQuery   : fetchBaseQuery({
        baseUrl : process.env.NODE_ENV === 'development' ? 'http://localhost:3001' : process.env.REACT_APP_BACKEND_ENDPOINT!,
    }),
    endpoints   : builder => ({
        getBundle    : builder.query<Bundle, BundleGetParams>({
            query             : params => ({
                url     : `/api/bundles/${params.bundleId!}`,
                method  : 'GET',
                headers : params.idToken ? {
                    Authorization : `Bearer ${params.idToken}`,
                } : {},
            }),
            transformResponse : (response : Record<string, any>) => ({
                id          : response.id,
                displayName : response.displayName,
                description : response.description,
                iconId      : response.iconId,
                isPrivate   : Boolean(response.isPrivate),
                viewCount   : Number(response.viewCount),
                createdBy   : response.createdBy,
                createdAt   : response.createdAt,
                updatedAt   : response.updatedAt,
                packages    : transformPackages(response.packages),
            }),
            extraOptions      : {
                maxRetries : API_MAX_RETRIES,
            },
        }),
        getBundles   : builder.query<Bundle[], BundleGetParams>({
            query             : params => ({
                url     : '/api/bundles',
                method  : 'GET',
                headers : params.idToken ? {
                    Authorization : `Bearer ${params.idToken}`,
                } : {},
            }),
            transformResponse : (response : Record<string, any>[]) => response.map(record => ({
                id          : record.id,
                displayName : record.displayName,
                description : record.description,
                iconId      : record.iconId,
                isPrivate   : Boolean(record.isPrivate),
                viewCount   : Number(record.viewCount),
                createdBy   : record.createdBy,
                createdAt   : record.createdAt,
                updatedAt   : record.updatedAt,
                packages    : transformPackages(record.packages),
            })),
            extraOptions      : {
                maxRetries : API_MAX_RETRIES,
            },
        }),
        createBundle : builder.mutation<void, BundleCreateParams>({
            query           : params => ({
                url             : '/api/bundles',
                method          : 'POST',
                body            : params.bundle,
                headers         : {
                    Authorization : `Bearer ${params.idToken}`,
                },
                responseHandler : async response => await response.text(),
            }),
            extraOptions    : {
                maxRetries : API_MAX_RETRIES,
            },
        }),
        updateBundle : builder.mutation<void, BundleUpdateParams>({
            query           : params => ({
                url     : `/api/bundles/${params.bundle.id}`,
                method  : 'PUT',
                body    : params.bundle,
                headers : {
                    Authorization : `Bearer ${params.idToken}`,
                },
                responseHandler : async response => await response.text(),
            }),
            extraOptions    : {
                maxRetries : API_MAX_RETRIES,
            },
        }),
        deleteBundle : builder.mutation<void, BundleDeleteParams>({
            query           : params => ({
                url             : `/api/bundles/${params.bundleId}`,
                method          : 'DELETE',
                headers         : {
                    Authorization : `Bearer ${params.idToken}`,
                },
                responseHandler : async response => await response.text(),
            }),
            extraOptions    : {
                maxRetries : API_MAX_RETRIES,
            },
        }),
    }),
});

export const { useCreateBundleMutation, useDeleteBundleMutation, useGetBundleQuery, useGetBundlesQuery, useUpdateBundleMutation, } = bundleApi;
