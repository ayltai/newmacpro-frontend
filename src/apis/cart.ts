import { createApi, fetchBaseQuery, } from '@reduxjs/toolkit/query/react';

import { API_MAX_RETRIES, } from '../constants';
import type { Cart, } from '../types';

export const cartApi = createApi({
    reducerPath : 'cartApi',
    baseQuery   : fetchBaseQuery({
        baseUrl : process.env.NODE_ENV === 'development' ? 'http://localhost:3001' : process.env.REACT_APP_BACKEND_ENDPOINT!,
    }),
    tagTypes    : [
        'Cart',
    ],
    endpoints   : builder => ({
        getCart : builder.query<Cart, string>({
            query        : id => `/api/carts/${id}`,
            extraOptions : {
                maxRetries : API_MAX_RETRIES,
            },
        }),
        addCart : builder.mutation<void, Cart>({
            query           : cart => ({
                url             : '/api/carts',
                method          : 'POST',
                body            : cart,
                responseHandler : async response => await response.text(),
            }),
            invalidatesTags : [
                'Cart',
            ],
            extraOptions    : {
                maxRetries : API_MAX_RETRIES,
            },
        }),
    }),
});

export const { useGetCartQuery, useAddCartMutation, } = cartApi;
