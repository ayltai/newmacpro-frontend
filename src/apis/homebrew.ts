import { createApi, fetchBaseQuery, } from '@reduxjs/toolkit/query/react';

import { API_MAX_RETRIES, } from '../constants';
import type { HomebrewCaskPackage, HomebrewCorePackage, Package, } from '../types';

export const homebrewApi = createApi({
    reducerPath : 'homebrewApi',
    baseQuery   : fetchBaseQuery({
        baseUrl : 'https://formulae.brew.sh',
    }),
    endpoints   : builder => ({
        cask : builder.query<Package[], void>({
            query             : () => '/api/cask.json',
            transformResponse : (response : HomebrewCaskPackage[]) => (response || []).map(result => ({
                id               : result.token,
                displayName      : result.name[0],
                description      : result.desc,
                version          : result.version,
                website          : result.homepage,
                monthlyDownloads : result.analytics?.install['30d'][result.token],
                source           : 'Homebrew/cask',
            })),
            extraOptions      : {
                maxRetries : API_MAX_RETRIES,
            },
        }),
        core : builder.query<Package[], void>({
            query             : () => '/api/formula.json',
            transformResponse : (response : HomebrewCorePackage[]) => (response || []).map(result => ({
                id               : result.name,
                displayName      : result.full_name ?? result.name,
                description      : result.desc,
                version          : result.versions.stable,
                license          : result.license,
                website          : result.homepage,
                monthlyDownloads : result.analytics?.install_on_request['30d'][result.name],
                source           : 'Homebrew/core',
            })),
            extraOptions      : {
                maxRetries : API_MAX_RETRIES,
            },
        }),
    }),
});

export const { useCaskQuery, useCoreQuery, } = homebrewApi;
