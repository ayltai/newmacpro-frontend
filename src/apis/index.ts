export { appStoreApi, useSearchQuery, } from './appStore';
export { bundleApi, useCreateBundleMutation, useDeleteBundleMutation, useGetBundleQuery, useGetBundlesQuery, useUpdateBundleMutation, } from './bundle';
export { cartApi, useAddCartMutation, useGetCartQuery, } from './cart';
export { homebrewApi, useCaskQuery, useCoreQuery, } from './homebrew';
export { tweakApi, useGetPlaybookQuery, useGetTweaksQuery, } from './tweak';
