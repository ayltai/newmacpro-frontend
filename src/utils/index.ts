import * as Sentry from '@sentry/react';

export const handleError = (error : any) => {
    if (process.env.NODE_ENV === 'production') {
        Sentry.captureException(error);
    } else {
        console.error(error);
    }
};

export const capitalizeFirstLetter = (input : string) => input.charAt(0).toUpperCase() + input.slice(1);
