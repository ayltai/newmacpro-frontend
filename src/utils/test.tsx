import { ThemeProvider, } from '@mui/material';
import { configureStore, Store, } from '@reduxjs/toolkit';
import { act, render, } from '@testing-library/react';
import mediaQuery from 'css-mediaquery';
import { ConfirmProvider, } from 'material-ui-confirm';
import React, { ReactNode, } from 'react';
import { Provider, } from 'react-redux';

import { bundleApi, cartApi, } from '../apis';
import { POPULAR_APPS, } from '../constants';
import { cartReducer, prefReducer, store as reduxStore, } from '../states';
import { AppTheme, } from '../styles';
import { Cart, } from '../types';

export const mockApis = async (cart? : Cart) => {
    fetchMock.doMock(request => {
        if (request.url.endsWith('/api/bundles') && request.method === 'POST') return Promise.resolve({
            status : 201,
            body   : 'Created',
        });

        if (request.url.endsWith('/api/bundles/1') && request.method === 'GET') return Promise.resolve({
            status : 200,
            body   : JSON.stringify({
                id          : '1',
                displayName : 'Dummy',
                packages    : [
                    POPULAR_APPS[0],
                ],
            }),
        });

        if (request.url.endsWith('/api/bundles/1') && request.method === 'DELETE') return Promise.resolve({
            status : 200,
            body   : 'OK',
        });

        if (request.url.endsWith('/api/carts') && request.method === 'POST') return Promise.resolve({
            status : 201,
            body   : 'Created',
        });

        return Promise.resolve({
            status : 404,
            body   : 'Not found',
        });
    });

    await act(async () => {
        await reduxStore.dispatch(bundleApi.endpoints?.getBundle.initiate({
            bundleId : '1',
            idToken  : 'idToken',
        }));
        if (cart) await reduxStore.dispatch(cartApi.endpoints.addCart.initiate(cart));
    });
};

export const createMatchMedia = (width : number) => (query : string) => ({
    matches             : mediaQuery.match(query, {
        width,
    }),
    media               : '',
    onchange            : null,
    addListener         : () => {},
    removeListener      : () => {},
    addEventListener    : () => {},
    removeEventListener : () => {},
    dispatchEvent       : () => false,
});

export const createStore = (preloadedState? : Record<string, any>) => configureStore({
    preloadedState,
    reducer    : {
        cart                      : cartReducer,
        pref                      : prefReducer,
        [ bundleApi.reducerPath ] : bundleApi.reducer,
        [ cartApi.reducerPath   ] : cartApi.reducer,
    },
    middleware : getDefaultMiddleware => getDefaultMiddleware().concat(bundleApi.middleware, cartApi.middleware),
});

export const defaultStore = createStore();

export type RootState = ReturnType<typeof defaultStore.getState>;

const customRender = (ui : any, {
    preloadedState,
    store = createStore(preloadedState),
    ...rest
} : {
    preloadedState?   : Record<string, any>,
    store?            : Store<RootState>,
    [ rest : string ] : any,
} = {}) => render(ui, {
    wrapper : ({
        children,
    } : {
        children : ReactNode,
    }) => (
        <Provider store={store}>
            <ThemeProvider theme={AppTheme}>
                <ConfirmProvider>
                    {children}
                </ConfirmProvider>
            </ThemeProvider>
        </Provider>
    ),
    ...rest,
});

export * from '@testing-library/react';

export { customRender as render, };
