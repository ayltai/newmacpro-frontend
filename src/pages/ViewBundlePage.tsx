import { Button, } from '@mui/material';
import { useConfirm, } from 'material-ui-confirm';
import React, { useEffect, useState, } from 'react';
import { useTranslation, } from 'react-i18next';

import { useDeleteBundleMutation, useGetBundleQuery, useUpdateBundleMutation, } from '../apis';
import { BundleDetails, Loading, Message, Page, PageActions, PageContent, PageTitle, } from '../components';
import type { Bundle, } from '../types';
import { handleError, } from '../utils';
import { PageProps, } from './Page.types';

export const ViewBundlePage = ({
    open,
    bundleId,
    idToken,
    onClose,
} : PageProps & {
    bundleId : string,
    idToken? : string,
}) => {
    const { data, refetch, error, isFetching, isLoading, } = useGetBundleQuery({
        bundleId,
        idToken,
    });

    const [ updateBundle, updateBundleResult, ] = useUpdateBundleMutation();
    const [ deleteBundle, deleteBundleResult, ] = useDeleteBundleMutation();

    const [ bundle,     setBundle,     ] = useState<Bundle | undefined>({
        id       : bundleId,
        packages : [],
    });
    const [ message,    setMessage,    ] = useState<string | undefined>();
    const [ hasError,   setHasError,   ] = useState(false);
    const [ hasMessage, setHasMessage, ] = useState(false);

    const confirm = useConfirm();

    const { t, } = useTranslation();

    const handleMessageClose = () => {
        setHasMessage(false);

        if (onClose) onClose();
    };

    const handleErrorClose = () => {
        setHasError(false);

        if (onClose) onClose();
    };

    const handleUpdate = async () => {
        if (idToken) await updateBundle({
            bundle : bundle!,
            idToken,
        });
    };

    const handleDelete = () => {
        if (idToken) confirm({
            title       : t('bundle.delete.confirm_title'),
            description : t('bundle.delete.confirm_description'),
        }).then(async () => {
            await deleteBundle({
                bundleId,
                idToken,
            });
        }).catch(() => {
            // Does nothing
        });
    };

    useEffect(() => {
        if (updateBundleResult.isError) {
            setMessage('message.bundle.update_failed');
            setHasMessage(true);
        } else if (updateBundleResult.isSuccess) {
            setMessage('message.bundle.updated');
            setHasMessage(true);

            refetch().catch(handleError);
        }
    }, [ updateBundleResult.isError, updateBundleResult.isSuccess, ]);

    useEffect(() => {
        if (deleteBundleResult.isError) {
            setMessage('message.bundle.create_failed');
            setHasMessage(true);
        } else if (deleteBundleResult.isSuccess) {
            setMessage('message.bundle.created');
            setHasMessage(true);

            refetch().catch(handleError);
        }
    }, [ deleteBundleResult.isError, deleteBundleResult.isSuccess, ]);

    useEffect(() => setBundle(data), [ data, ]);

    useEffect(() => {
        if (error) {
            setHasError(true);

            handleError(error);
        }
    }, [ error, ]);

    return (
        <>
            <Page
                open={open}
                onClose={onClose}>
                <PageTitle>
                    {t('bundle.load.title')}
                </PageTitle>
                <PageContent>
                    {bundle && (
                        <BundleDetails
                            readOnly={!idToken}
                            bundle={bundle}
                            onChange={setBundle} />
                    )}
                </PageContent>
                <PageActions sx={{
                    display : idToken ? 'flex' : 'none',
                }}>
                    <Button onClick={handleUpdate}>{t('action.update_bundle')}</Button>
                    <Button onClick={handleDelete}>{t('action.delete_bundle')}</Button>
                </PageActions>
            </Page>
            <Loading
                show={!hasMessage && (isLoading || isFetching)}
                message={t('bundle.load.in_progress')} />
            <Message
                open={hasMessage}
                message={message ? t(message) : ''}
                onClose={handleMessageClose} />
            <Message
                open={hasError}
                message={t('message.bundle.load_failed')}
                onClose={handleErrorClose} />
        </>
    );
};
