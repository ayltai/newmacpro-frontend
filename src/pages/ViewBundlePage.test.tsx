import React from 'react';

import { bundleApi, } from '../apis';
import { fireEvent, mockApis, render, waitFor, } from '../utils/test';
import { ViewBundlePage, } from './ViewBundlePage';

describe('<ViewBundlePage />', () => {
    beforeEach(async () => {
        await mockApis();
    });

    it('renders correctly', async () => {
        const component = render(
            <ViewBundlePage
                open
                bundleId='1' />
        );

        await waitFor(() => {
            expect(component).toMatchSnapshot();
        });
    });

    it('hides Update and Delete buttons for guest user', async () => {
        const { queryAllByRole, } = render(
            <ViewBundlePage
                open
                bundleId='1' />
        );

        await waitFor(() => {
            expect(queryAllByRole('button')).toHaveLength(0);
        });
    });

    it('shows Update and Delete buttons for authenticated user', async () => {
        const { queryAllByRole, } = render(
            <ViewBundlePage
                open
                bundleId='1'
                idToken='idToken' />
        );

        await waitFor(() => {
            expect(queryAllByRole('button')).toHaveLength(2);
            expect(queryAllByRole('button')[0]).toHaveTextContent('action.update_bundle');
            expect(queryAllByRole('button')[1]).toHaveTextContent('action.delete_bundle');
        });
    });

    it('calls Update Bundle API when Update button is clicked', async () => {
        const initiate = jest.spyOn(bundleApi.endpoints?.updateBundle, 'initiate');

        const { getAllByRole, } = render(
            <ViewBundlePage
                open
                bundleId='1'
                idToken='idToken' />
        );

        await waitFor(() => {
            fireEvent.click(getAllByRole('button')[0]);
        });

        expect(initiate).toHaveBeenCalledTimes(1);
    });

    it('calls Delete Bundle API when Delete button is clicked', async () => {
        const initiate = jest.spyOn(bundleApi.endpoints?.deleteBundle, 'initiate');

        const { getAllByRole, } = render(
            <ViewBundlePage
                open
                bundleId='1'
                idToken='idToken' />
        );

        await waitFor(() => {
            fireEvent.click(getAllByRole('button')[1]);
        });

        expect(initiate).toHaveBeenCalledTimes(1);
    });
});
