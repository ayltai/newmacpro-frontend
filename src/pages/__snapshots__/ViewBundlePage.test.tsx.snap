// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewBundlePage /> renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root css-sox5kk-MuiBackdrop-root"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        aria-hidden="true"
        class="MuiBackdrop-root css-sox5kk-MuiBackdrop-root"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiBox-root css-xi606m"
        >
          <span
            class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorSecondary css-gppelf-MuiCircularProgress-root"
            role="progressbar"
            style="width: 40px; height: 40px;"
          >
            <svg
              class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1kriq04-MuiTypography-root"
          >
            bundle.load.in_progress
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiModal-root MuiDialog-root css-29ydf2-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-1sep8xo-MuiDialog-container"
        role="presentation"
        style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r0:"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullScreen css-hlf5l-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-jp4g6a-MuiTypography-root-MuiDialogTitle-root"
            id=":r0:"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1ywgf30-MuiTypography-root"
            >
              bundle.load.title
            </h6>
          </h2>
          <div
            class="MuiDialogContent-root MuiDialogContent-dividers css-1t4vnk2-MuiDialogContent-root"
          >
            <div
              class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
            >
              <div
                class="MuiStack-root css-1sazv7p-MuiStack-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-1eqzuua-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for=":r4:"
                    id=":r4:-label"
                  >
                    bundle.create.display_name
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-1wv3h1b-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      disabled=""
                      id=":r4:"
                      required=""
                      type="text"
                      value="Dummy"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-yjsfm1"
                      >
                        <span>
                          bundle.create.display_name
                           
                          *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <div
                  class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-1eqzuua-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for=":r5:"
                    id=":r5:-label"
                  >
                    bundle.create.description
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-1wv3h1b-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      disabled=""
                      id=":r5:"
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-yjsfm1"
                      >
                        <span>
                          bundle.create.description
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                />
                <h6
                  class="MuiTypography-root MuiTypography-h6 css-1bljjaj-MuiTypography-root"
                >
                  bundle.create.items
                </h6>
                <ul
                  class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
                />
              </div>
            </div>
          </div>
          <div
            class="MuiDialogActions-root MuiDialogActions-spacing css-14zwa3a-MuiDialogActions-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-15b00zv-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              action.update_bundle
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-15b00zv-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              action.delete_bundle
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root css-sox5kk-MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      aria-hidden="true"
      class="MuiBackdrop-root css-sox5kk-MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiBox-root css-xi606m"
      >
        <span
          class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorSecondary css-gppelf-MuiCircularProgress-root"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </span>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1kriq04-MuiTypography-root"
        >
          bundle.load.in_progress
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
